runtime/vm/OutOfLineINL_openj9_internal_foreign_abi_InternalDowncallHandler.cpp

```diff
@@ -141,8 +146,13 @@ OutOfLineINL_openj9_internal_foreign_abi_InternalDowncallHandler_initCifNativeTh
 				goto freeAllMemoryThenExit;
 			/* Only intended for struct as the primitive's ffi_type is non-null */
 			} else if ((NULL == argTypes[argIndex])
-			|| ((FFI_TYPE_STRUCT == argTypes[argIndex]->type) && (NULL == argTypes[argIndex]->elements))
-			) {
+						|| (((FFI_TYPE_STRUCT == argTypes[argIndex]->type)
+#if defined(J9ZOS390)
+							|| (FFI_TYPE_STRUCT_FF == returnType->type)
+							|| (FFI_TYPE_STRUCT_DD == returnType->type)
```

Comment:
Should these use `argTypes[argIndex]` instead of `returnType`?
I'm not sure the parentheses are nested properly.
Line 145 should not change:
```c
			) {
```

Reply:
Fixed in https://github.com/eclipse-openj9/openj9/pull/19645/commits/38245ec2872c2b2664f6fdd3947f80e71e63c57d